paddle/scripts/teamcity/pr_build.sh

```diff
@@ -0,0 +1,115 @@
+#!/bin/bash
+#
+# This build script is used for TeamCity CI, we need to 
+# set the follwoing parameters in the project configuration:
+#   PARAMETER            | EXAMPLE    |   COMMENTS
+# CTEST_OUTPUT_ON_FAILURE   1         // whether output failure message
+# CTEST_PARALLEL_LEVEL      5         // parallel level for running ctest
+# PADDLE_VERSION            0.11.0    // the latest tag
+# WITH_AVX                  ON        // wheter the instruction support AVX
+# WITH_GOLANG               OFF       // don't build the Golang code
+# WITH_GPU                  ON        // running on GPU
+# WITH_UBUNTU_MIRROR        ON        // use unbuntu mirro to speed up dev Docker image building
+#
+
+set -xe
+echo $(env)
+PR_ID=$(echo %teamcity.build.vcs.branch.Paddle_PaddlePaddle% | sed 's/[^0-9]*//g')
+echo ${PR_ID}
+
+CHANGED_FILES=$(cat %system.teamcity.build.changedFiles.file% | awk -F : '{print $1}')
+if [ -z "$CHANGED_FILES" ]
+then
+  echo "empty changed files"
+else
+  echo "changed files: $CHANGED_FILES"
+  COUNT=$(echo $CHANGED_FILES | awk -F . '{print $NF}'|grep -i -E 'md|rst|jpg|png' | wc -l)
+  FILE_COUNT=$(echo $CHANGED_FILES | wc -l)
+  if [ "$COUNT" -eq "$FILE_COUNT" ]
+  then
+    echo "all readme files or pictures, skip CI"
+    exit 0
+  fi
+fi
+
+apt_mirror='s#http://archive.ubuntu.com/ubuntu#mirror://mirrors.ubuntu.com/mirrors.txt#g'
+
+export PADDLE_DEV_NAME=paddlepaddle/paddle:dev
+
+nvidia-docker run -i --rm -v $PWD:/paddle ${PADDLE_DEV_NAME} rm -rf /paddle/build
+
+# Use the cached thirdparty package if the MD5 is mathes.
+md5_content=$(cat \
+              ${PWD}/cmake/external/*.cmake \
+              ${PWD}/paddle/scripts/teamcity/pr_build.sh \
```

Comment:
Maybe don't need this line anymore it we don't check-in `pre-build.sh` to our repo.

Reply:
Done.